src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,169 @@
 package core.basesyntax;
 
+import java.util.Arrays;
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_INITIAL_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int DOUBLING_FACTOR = 2;
+    private Node<K,V>[] table;
+    private int threshold;
+    private int size;
+
+    @SuppressWarnings({"unchecked"})
+    public MyHashMap() {
+        this.table = (Node<K,V>[]) new Node[DEFAULT_INITIAL_CAPACITY];
+        this.size = 0;
+        this.threshold = (int) (DEFAULT_INITIAL_CAPACITY * DEFAULT_LOAD_FACTOR);
+    }
 
     @Override
     public void put(K key, V value) {
-
+        if (size >= threshold) {
+            threshold *= DOUBLING_FACTOR;
+            resize();
+        }
+        Node<K, V> newNode = new Node<>(hash(key), key, value, null);
+        int index = (table.length - 1) & newNode.hash;
+        if (table[index] == null) {
+            justAddNewNode(index, newNode);
+        } else {
+            Node<K, V> current = table[index];
+            while (current.next != null) {
+                if (theseKeysAreEqual(current.key, newNode.key)) {
+                    rewrightValue(current, newNode);
```

Comment:
The method name `rewrightValue` seems to be a typo. It should be `rewriteValue` to correctly represent its functionality.